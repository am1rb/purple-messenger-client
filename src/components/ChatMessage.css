.holder
{
	/*width: 200px;*/
}

.message
{
	padding: 10px;
	margin: 15px;
	position: relative;
	
	max-width: 500px;
	/*flex-basis: 60%;
    max-width: 60%;*/

    /*overflow: hidden;*/
	
}


.message_me
{
	float: right;
	background: #e2ffc7;
}


.text
{
	margin-bottom: 5px;
	word-wrap: break-word;
}

.message_friend
{
	float: left;	
	background: #fff;
}

.clr
{
	display: block;
	clear: both;
}

.date
{
	display: inline-block;
	font-size: 0.9em;
}

.message_me .date
{
	color: #5ca04a;
}

.message_friend
{
	float: left;	
	background: #fff;
}


.message_friend .date
{
	color: #a1aab3;
}

/*

	

	


	.textarea
	{
		height: 15vh;	
		width: 100%;
		direction: rtl;
		text-align: right;
		font-family: Vazir;
		padding: 10px;
	}

	.send-button
	{
		height: 5vh;	
		width: 100%;
	}
 */