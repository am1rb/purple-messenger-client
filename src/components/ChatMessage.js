// @flow

import React from 'react'
import classNames from 'classnames'

import ChatMessageStatus from './ChatMessageStatus'

import styles from './ChatMessage.css'

type Props = {
	date: string,
	status: 'received' | 'pending' | 'sent',
	isFriend: boolean,
	message: string,
};


export default class ChatMessage extends React.PureComponent<Props>
{
	render()
	{
		const classes = classNames({
			[styles.message]: styles.message,
			[styles.message_me]: !this.props.isFriend,
			[styles.message_friend]: this.props.isFriend,
		})

		return (
			<div className={styles.holder}>
				<div className={classes}>
					<div className={styles.text}>
						{this.props.message}
					</div>
					<div className={styles.detail}>

						{!this.props.isFriend && <ChatMessageStatus status={this.props.status} />}

						<span className={styles.date}>
							{this.props.date}
						</span>

					</div>
				</div>
				<div className={styles.clr} />
			</div>
		)
	}
}