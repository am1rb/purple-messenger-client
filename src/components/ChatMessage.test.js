// @flow

import React from 'react'
import renderer from 'react-test-renderer'

import ChatMessage from './ChatMessage'


describe("<ChatMessage /> component tests.", () => {

	it("Should not regress", () => {

		const component = renderer.create(
			<ChatMessage
				date={'Tue Oct 02 2018 20:47:36'}
				isFriend={true}
				message="This is test"
				status='pending'
			/>
		)

		expect(component.toJSON()).toMatchSnapshot()

	})

})