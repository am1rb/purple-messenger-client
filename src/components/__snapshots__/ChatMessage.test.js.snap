// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatMessage /> component tests. Should not regress 1`] = `
<div
  className={undefined}
>
  <div
    className="undefined"
  >
    <div
      className={undefined}
    >
      This is test
    </div>
    <div
      className={undefined}
    >
      <span
        className={undefined}
      >
        Tue Oct 02 2018 20:47:36
      </span>
    </div>
  </div>
  <div
    className={undefined}
  />
</div>
`;
